test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -2061,6 +2061,17 @@ public Task HEADERS_Received_HeaderBlockContainsTEHeader_ValueIsNotTrailers_Stre
 
             await SendGoAwayAsync();
 
+            var spinLimit = 50; // 2.5 seconds
+            while (_connection.State == Http2Connection.ConnectionState.Open && spinLimit --> 0)
+            {
+                await Task.Delay(50);
+            }
+            Assert.Equal(Http2Connection.ConnectionState.Closing, _connection.State);
+
+            await SendRstStreamAsync(1);
+            await SendRstStreamAsync(3);
+            await SendRstStreamAsync(5);
+
             await WaitForConnectionStopAsync(expectedLastStreamId: 5, ignoreNonGoAwayFrames: true);
```

Comment:
ignoreNonGoAwayFrames: false, read each expected response.

Reply:
:+1: This was there before. I haven't double check why, but generally speaking you should have a very good reason to use `ignoreNonGoAwayFrames: true`. It's usually much better to assert exactly which frames you expect.